test/gpu/native/reduction/basic.chpl

```diff
@@ -32,7 +33,9 @@ proc testType(type t) {
         when "maxloc" do res=gpuMaxLocReduce(Arr);
       }
 
-      writeln(op, ": ", res);
+      // we cast to `t` to paper over the difference between the integral values
+      // returned from two different ways of sum-reducing bool arrays
+      writeln(op, ": ", if isTuple(res) then res:(t,int) else res:t);
```

Comment:
what are the two different ways?

Reply:
`gpuSumReduce(MyBoolArr)` and `+ reduce MyBoolArr`. I'll clarify the comment.

Reply:
Where the former will return a bool that can be either 1 or 0. The latter will literally sum up the integral values stored in bools (which is what Chapel does today for other `reduce` expressions, and considered correct for the scope of this PR) and return a bool with that integral value.